sidekick_core/lib/src/commands/format_command.dart

```diff
@@ -0,0 +1,203 @@
+import 'package:glob/glob.dart';
+import 'package:glob/list_local_fs.dart';
+import 'package:meta/meta.dart';
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:yaml/yaml.dart';
+
+/// Formats the Code for all Flutter/Dart packages in the repository
+class FormatCommand extends Command {
+  @override
+  String get description => 'Formatting the code for all packages';
+
+  @override
+  String get name => 'format';
+
+  /// packages whose files should not be formatted
+  final List<DartPackage> exclude;
+
+  /// glob patterns of packages whose dependencies should not be loaded
+  ///
+  /// Search starts at repository root.
+  ///
+  /// Example project layout:
+  ///
+  /// ```
+  /// repo-root
+  /// ├── packages
+  /// │   ├── package1
+  /// │   ├── package2
+  /// │   └── circle
+  /// └── third_party
+  ///     ├── circle
+  ///     │   ├── packageA
+  ///     │   └── packageB
+  ///     └── square
+  /// ```
+  ///
+  /// - Use `packages/package1/**` to exclude only `packages/package1`.
+  /// - Use `**/circle/**` to exclude `packages/circle` as well as
+  ///   `third_party/circle/packageA` and `third_party/circle/packageB`.
+  final List<String> excludeGlob;
+
+  FormatCommand({
+    this.exclude = const [],
+    this.excludeGlob = const [],
+  }) {
+    argParser.addOption(
+      'package',
+      abbr: 'p',
+    );
+    argParser.addOption(
+      'line-length',
+      abbr: 'l',
+    );
+    argParser.addFlag(
+      'verify',
+      help:
+          'Only verifies that all code is formatted, does not actually format it',
+    );
+  }
+
+  @override
+  Future<void> run() async {
+    final String? packageName = argResults?['package'] as String?;
+    final int? lineLength =
+        int.tryParse(argResults?['line-length'] as String? ?? '');
+    final bool verify = argResults?['verify'] as bool? ?? false;
+
+    final root = SidekickContext.repository ?? SidekickContext.projectRoot;
+    final List<DartPackage> allPackages = findAllPackages(root);
+    if (packageName != null) {
+      final package =
+          allPackages.where((it) => it.name == packageName).firstOrNull;
+      if (package == null) {
+        throw "Package with name $packageName not found in repository "
+            "${SidekickContext.repository?.path}";
+      }
+      // only format for selected package
+      // _format(package, globalLineLength: lineLength);
+      return;
+    }
+    final globExcludes = excludeGlob
+        .expand((rule) {
+          // start search at repo root
+          final root = SidekickContext.repository?.path;
```

Comment:
fallback to `null` would use the current directory, which would be wrong. Use `SidekickContext.projectRoot`